### ISSUE TITLE
Flow is complaining "Cannot reassign constant"

### ISSUE THREAD

**Issue by wanderley** [nvie/decoders#111]
Thanks for your work here! I love Elm Decoders/Encoders and I was missing it for Flow. I added this library in my project and flow is complaining about constants that are reassigned. 

```
Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ node_modules/debrief/lib/serialize.js.flow:15:5

Cannot reassign constant s [1].

 [1]  6│ function serializeString(s: string, width: number = 80) {
       :
     12│     }
     13│
     14│     // Cut off a bit
     15│     s = s.substring(0, width - 15) + '...';
     16│     ser = JSON.stringify(s) + ' [truncated]';
     17│     return ser;
     18│ }


Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ node_modules/decoders/lib/either.js.flow:13:5

Cannot reassign constant s [1].

     10│  */
     11│ // istanbul ignore next
 [1] 12│ function itemize(s: string = ''): string {
     13│     s = indent(s);
     14│     return '-' + s.substring(1);
     15│ }
     16│



Found 2 errors
```
I am using Flow 0.72.0 in [strict local mode](https://flow.org/en/docs/strict/#strict-local-).

**Comment by wanderley** [comment]
I thought it was caused by the strict mode but it seems to be related with `experimental.const_params` which is turned on in my project. I will investigate a bit more before sending a PR. Thanks for you fast response!

**Comment by nvie** [comment]
Thanks @wanderley – this fix is now published in [v1.6.1](https://www.npmjs.com/package/decoders).

**Comment by wanderley** [comment]
Thanks @nvie!
